neoemf-core/src/main/java/fr/inria/atlanmod/neoemf/core/internal/LazyStoreFeatureMap.java

```diff
@@ -0,0 +1,250 @@
+     * @param feature the feature associated with this list
+     */
+    public LazyStoreFeatureMap(PersistentEObject owner, EStructuralFeature feature) {
+        super(owner, feature);
+        this.storable = owner;
+    }
+
+    @Nonnull
+    protected StoreAdapter eStore() {
+        return storable.eStore();
+    }
+
+    // region Delegating methods (identical to LazyStoreList)
+
+    @Override
+    protected List<Entry> delegateList() {
+        throw new UnsupportedOperationException("delegateList");
+    }
+
+    @Override
+    protected int delegateSize() {
+        return eStore().size(owner, eStructuralFeature);
+    }
+
+    @Override
+    protected boolean delegateIsEmpty() {
+        return eStore().isEmpty(owner, eStructuralFeature);
+    }
+
+    @Override
+    protected boolean delegateContains(Object object) {
+        return eStore().contains(owner, eStructuralFeature, object);
+    }
+
+    @Override
+    protected boolean delegateContainsAll(Collection<?> collection) {
+        if (isNull(collection) || collection.isEmpty()) {
+            return false;
+        }
+
+        if (collection.size() == 1) {
+            return contains(collection.iterator().next());
+        }
+
+        return eStore().getAll(owner, eStructuralFeature).containsAll(collection);
+    }
+
+    @Override
+    protected int delegateIndexOf(Object object) {
+        return eStore().indexOf(owner, eStructuralFeature, object);
+    }
+
+    @Override
+    protected int delegateLastIndexOf(Object object) {
+        return eStore().lastIndexOf(owner, eStructuralFeature, object);
+    }
+
+    @Override
+    protected Object[] delegateToArray() {
+        return eStore().toArray(owner, eStructuralFeature);
+    }
+
+    @Override
+    protected <T> T[] delegateToArray(T[] array) {
+        return eStore().toArray(owner, eStructuralFeature, array);
+    }
+
+    @Override
+    protected Entry delegateGet(int index) {
+        return (Entry) eStore().get(owner, eStructuralFeature, index);
+    }
+
+    @Override
+    protected Entry delegateSet(int index, Entry object) {
+        return (Entry) eStore().set(owner, eStructuralFeature, index, object);
+    }
+
+    @Override
+    protected void delegateAdd(Entry object) {
+        delegateAdd(InternalEObject.EStore.NO_INDEX, object);
+    }
+
+    @Override
+    protected void delegateAdd(int index, Entry object) {
+        eStore().add(owner, eStructuralFeature, index, object);
+    }
+
+    @Override
+    protected Entry delegateRemove(int index) {
+        return (Entry) eStore().remove(owner, eStructuralFeature, index);
+    }
+
+    @Override
+    protected void delegateClear() {
+        eStore().clear(owner, eStructuralFeature);
+    }
+
+    @Override
+    protected Entry delegateMove(int targetIndex, int sourceIndex) {
+        return (Entry) eStore().move(owner, eStructuralFeature, targetIndex, sourceIndex);
+    }
+
+    @Override
+    // FIXME Does not resolve
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Take the required action to fix the issue indicated by this comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1134)